src/Temporalio/Workflows/WorkflowDefinition.cs

```diff
@@ -286,6 +304,27 @@ public static WorkflowDefinition Create(
                         runMethod = method;
                     }
                 }
+                if (method.IsDefined(typeof(WorkflowDynamicOptionsAttribute), false))
+                {
+                    if (!method.IsPublic)
+                    {
+                        throw new ArgumentException($"WorkflowDynamicOptions method {method} must be public");
+                    }
+                    if (method.IsStatic)
+                    {
+                        throw new ArgumentException($"WorkflowDynamicOptions method {method} cannot be static");
+                    }
+                    if (method.ReturnType != typeof(WorkflowDefinitionOptions))
+                    {
+                        throw new ArgumentException($"WorkflowDynamicOptions method {method} must return WorkflowDefinitionOptions");
+                    }
+                    var parameters = method.GetParameters();
+                    if (parameters.Length != 0)
+                    {
+                        throw new ArgumentException($"WorkflowDynamicOptions method {method} must take no parameters");
+                    }
+                    dynamicOptionsMethod = instance => (WorkflowDefinitionOptions)method.Invoke(instance, null)!;
```

Comment:
Add one more check here that makes sure `dynamicOptionsMethod` isn't already set (i.e. we only allow one of these)